test/unit/classDecorator.spec.ts

```diff
@@ -0,0 +1,167 @@
+import * as util from "../util";
+
+test("Class decorator with no parameters", () => {
+    const source = `
+    function SetBool<T extends { new(...args: any[]): {} }>(constructor: T) {
+        return class extends constructor {
+            decoratorBool = true;
+        }
+    }
+
+    @SetBool
+    class TestClass {
+        public decoratorBool = false;
+    }
+
+    const classInstance = new TestClass();
+    return classInstance.decoratorBool;
+    `;
+
+    const result = util.transpileAndExecute(source);
+    expect(result).toBe(true);
+});
+
+test("Class decorator with parameters", () => {
+    const source = `
+    function SetNum(this: void, numArg: number) {
```

Comment:
Check if decorator factories work without `this: void`, they should work with and without it.